roofit/hs3/inc/RooFitHS3/RooJSONFactoryWSTool.h

```diff
@@ -59,27 +71,74 @@ class RooJSONFactoryWSTool {
       std::map<std::string, RooAbsArg *> objects;
    };
    mutable Scope _scope;
+   const RooFit::Detail::JSONNode *_rootnode_input = nullptr;
+   RooFit::Detail::JSONNode *_rootnode_output = nullptr;
+
+   RooFit::Detail::JSONNode &orootnode();
+   const RooFit::Detail::JSONNode &irootnode() const;
 
    RooWorkspace *_workspace;
+
+   static bool _stripObservables;
+
    static ImportMap _importers;
    static ExportMap _exporters;
-   void prepare();
+   static ExportKeysMap _exportKeys;
+   static ImportExpressionMap _pdfFactoryExpressions;
+   static ImportExpressionMap _funcFactoryExpressions;
+
    std::map<std::string, RooAbsData *> loadData(const RooFit::Detail::JSONNode &n);
    RooDataSet *unbinned(RooDataHist *hist);
    RooRealVar *getWeightVar(const char *name);
    RooRealVar *createObservable(const std::string &name, const RooJSONFactoryWSTool::Var &var);
 
 public:
+   static bool getStripObservables();
+   static void setStripObservables(bool);
+
+   class MissingRootnodeError : public std::exception {
+   public:
+      virtual const char *what() const noexcept override { return "no rootnode set"; }
+   };
+
+   class DependencyMissingError : public std::exception {
+      std::string _parent, _child, _class, _message;
```

Comment:
```suggestion
      std::string _parent;
      std::string _child;
      std::string _class
      std::string _message;
```
Try to avoid these comma lists of same type, it's clearer to write it out I think